php/WC_Gateway_Blockonomics.php

```diff
@@ -108,7 +108,15 @@ public function process_payment($order_id)
         $cancel_url = add_query_arg('order_key', $order->get_order_key(), $cancel_url);
 
         $blockonomics = new Blockonomics;
-        $responseObj = $blockonomics->new_address(get_option("blockonomics_callback_secret"));
+		$time_period = get_option("blockonomics_timeperiod", 10) *60;
+		$currentOrder = $blockonomics_orders[$order_id];
+		if(isset($currentOrder[$address])) {
+			if(time() > $timestamp + $time_period) {
+				$responseObj = $blockonomics->new_address(get_option("blockonomics_callback_secret"),1); 
```

Comment:
Using reset=1 will return the last generated address for the wallet. This needs to instead reuse the previous address linked to the order.

Reply:
removed all the parameters, and reverted from using the reset parameter.